corehq/apps/campaign/static/campaign/js/dashboard.js

```diff
@@ -222,8 +224,114 @@ var ReportWidget = function (config) {
 
     self.init = function () {
-        // TODO: Pass `config.report_options` to `getStandardHQReport()`
-        getStandardHQReport();
+
+        // Copied from `userreports/js/bootstrap5/base.js`
+        $(function () {
+            var chartSpecs = initialPageData.get('charts');
+            var updateCharts = function (data) {
+                if (chartSpecs !== null && chartSpecs.length > 0) {
+                    var isReportBuilderReport = initialPageData.get('created_by_builder');
+                    if (data.iTotalRecords > 25 && isReportBuilderReport) {
+                        $("#chart-warning").removeClass("hide");
+                        charts.clear($("#chart-container"));
+                    } else {
+                        $("#chart-warning").addClass("hide");
+                        charts.render(chartSpecs, data.aaData, $("#chart-container"));
+                    }
+                }
+            };
+
+            var mapSpec = initialPageData.get('map_config');
+            var updateMap = function (data) {
+                if (mapSpec) {
+                    mapSpec.mapboxAccessToken = initialPageData.get('MAPBOX_ACCESS_TOKEN');
+                    maps.render(mapSpec, data.aaData, $("#map-container"));
+                }
+            };
+
+            var paginationNotice = function (data) {
+                if (mapSpec) {  // Only show warning for map reports
+                    if (data.aaData !== undefined && data.iTotalRecords !== undefined) {
+                        if (data.aaData.length < data.iTotalRecords) {
+                            $('#info-message').html(
+                                gettext('Showing the current page of data. Switch pages to see more data.'),
+                            );
+                            $('#report-info').removeClass('hide');
+                        } else {
+                            $('#report-info').addClass('hide');
+                        }
+                    }
+                }
+            };
+
+            var errorCallback = function (jqXHR, textStatus, errorThrown) {
+                $('#error-message').html(errorThrown);
+                $('#report-error').removeClass('hide');
+            };
+
+            var successCallback = function (data) {
+                if (data.error || data.error_message) {
+                    const message = data.error || data.error_message;
+                    $('#error-message').html(message);
+                    $('#report-error').removeClass('hide');
+                } else {
+                    $('#report-error').addClass('hide');
+                }
+                if (data.warning) {
+                    $('#warning-message').html(data.warning);
+                    $('#report-warning').removeClass('hide');
+                } else {
+                    $('#report-warning').addClass('hide');
+                }
+            };
+
+            var reportTables = dataTablesConfig.HQReportDataTables({
```

Comment:
nit: are we supporting anything other than tables right now?
just thinking of reducing the copy here.

Reply:
Actually, I'd really like to remove this entire block of code, and instead fix [the import here](https://github.com/dimagi/commcare-hq/blob/2aee63ef99e1dcb0ec9e04eaf5941231a1c71814/corehq/apps/campaign/static/campaign/js/dashboard.js#L9).

I think it's happening because `dashboard.js` is using ES6 import syntax, but `userreports/js/bootstrap5/base` is still on `hqDefine` imports. What I need to do to fix this, is switch `userreports/js/bootstrap5/base` to use ES6 imports, as @biyeun did for [reports/js/bootstrap5/datatables_config.js](https://github.com/dimagi/commcare-hq/blob/dd4e41cd271bfda79aba0509df1709dd5d51473c/corehq/apps/reports/static/reports/js/bootstrap5/datatables_config.js#L1-L9). (@biyeun please correct me if I'm wrong.)

Reply:
That would be nice.

Reply:
0ddf91ad6d0dbda9c72abb3a0d803549ce31bec3